### ISSUE TITLE
Format selection (and thus format-on-paste) causes destructive edits

### ISSUE THREAD

**Issue by richardbullin** [microsoft/TypeScript#50076]
Issue Type: <b>Bug</b>

Create a simple template string in a Javascript or Typescript file as below
```javascript
const TEST_STRING = `${TEST_VAR_1}, this is some text`; 
```

Copy any text and paste anywhere after the closing expression bracket "}". All text including the last "}" is deleted. 

In my example I've copied the word "TEST_STRING " by double clicking on it. I've tried pasting  just before the end back tick

```javascript
const TEST_STRING = `${TEST_VAR_1; 
```
If you press undo the expected initial result is shown.

```javascript
const TEST_STRING = `${TEST_VAR_1}, this is some textTEST_STRING`; 
```

VS Code version: Code 1.69.2 (3b889b090b5ad5793f524b5d1d39fda662b96a2a, 2022-07-18T16:12:52.460Z)
OS version: Windows_NT x64 10.0.19043
Restricted Mode: No

Extensions: none

**Comment by mjbvz** [comment]
Confirmed with `^4.8.0-dev.20220727`. Requires `editor.formatOnPaste` to be enabled

Here's the sequence of events for pasting `abc`:

```
[Trace  - 23:33:55.294] <semantic> Sending request: updateOpen (72). Response expected: yes. Current queue length: 0
Arguments: {
    "changedFiles": [
        {
            "fileName": "/Users/matb/projects/san/index.js",
            "textChanges": [
                {
                    "newText": "abc",
                    "start": {
                        "line": 1,
                        "offset": 35
                    },
                    "end": {
                        "line": 1,
                        "offset": 35
                    }
                }
            ]
        }
    ],
    "closedFiles": [],
    "openFiles": []
}
[Trace  - 23:33:55.294] <semantic> Response received: updateOpen (72). Request took 0 ms. Success: true 
Result: true
[Trace  - 23:33:55.295] <syntax> Response received: updateOpen (28). Request took 1 ms. Success: true 
Result: true
[Trace  - 23:33:55.295] <syntax> Sending request: format (29). Response expected: yes. Current queue length: 0
Arguments: {
    "file": "/Users/matb/projects/san/index.js",
    "line": 1,
    "offset": 35,
    "endLine": 1,
    "endOffset": 38
}
[Trace  - 23:33:55.296] <syntax> Response received: format (29). Request took 2 ms. Success: true 
Result: [
    {
        "start": {
            "line": 1,
            "offset": 34
        },
        "end": {
            "line": 1,
            "offset": 96
        },
        "newText": ""
    }
]
```

**Comment by mjbvz** [comment]
Seems very similar to #48006

**Comment by amcasey** [comment]
> Seems very similar to #48006

+1.  @sandersn ?

**Comment by DanielRosenwasser** [comment]
Simpler repro is

```ts
const x = `${0}abc`;
```

Select `abc` and run format selection.

Note that:

1. This repro must have a template string with a substitution
2. The template tail literal must have some string contents.
3. There MUST be a semicolon at the end of the current statement.

**Comment by DanielRosenwasser** [comment]
Another similar one

```ts
foo(`${0} `)
```

1. This repro must have a template string with a substitution.
2. The template tail literal must have some string contents.
3. The template must be in a call.

**Comment by DanielRosenwasser** [comment]
First appears on `typescript@4.6.0-dev.20221130` which is hard to test because VS Code assumes it can run with IPC. That means it's one of https://github.com/microsoft/TypeScript/pulls?q=is%3Apr+merged%3A2021-11-29+is%3Aclosed, and likely https://github.com/microsoft/TypeScript/pull/46875.

I think we should get it fixed for 4.8, but it's been in the product for over 8 months so I guess I feel weird labeling it "High Priority".
